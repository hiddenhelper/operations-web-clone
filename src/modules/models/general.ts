import { Status, CompanyStatus } from './resource';
import { CompanyRole } from './project';
import { VisitorEntity } from './badge';
import { DeviceType } from './device';
import { IFormRules } from 'utils';

export type IEntityMap<T> = { [id: string]: T };

export interface ITokenResponse {
  email: string;
}

export interface ILoadingStatus {
  isLoading: boolean;
  hasError: boolean;
  error: any;
  traceId?: string;
}

export type IRelationUiMap = Record<string, any>;

export interface IPagination<T> {
  items: T[];
  pageNumber: number;
  totalResults: number;
  pageSize: number;
}

export interface IQueryParams {
  id?: string;
  page?: number;
  limit?: number;
  pageSize?: number;
  pageNumber?: number;
  query?: string;
  companyId?: string;
  excludeFromProjectId?: string;
  isDeveloper?: boolean;
  excludeProjectId?: string;
  stateCode?: string;
  countryCode?: string;
  period?: TimeFilterType;
  status?: Status | number;
  newPage?: boolean;
  filter?: number | string;
  filterByWorker?: boolean;
  nameContains?: string;
  maxItems?: number;
  projectId?: string;
  excludeCompanies?: string[];
  projectStatuses?: Status[];
  companyStatuses?: CompanyStatus[];
  isPaid?: boolean;
  sortType?: string;
  sortingName?: string;
  clientId?: string;
  tab?: string;
  active?: boolean | string;
  available?: boolean;
  type?: number;
  onlyPending?: boolean;
  role?: CompanyRole;
  entityType?: VisitorEntity;
  entity?: number;
  deviceType?: DeviceType;
  lastPage?: number;
  noRelatedProject?: boolean;
  name?: string;
  wStateCode?: string;
  wCountryCode?: string;
  wPeriod?: TimeFilterType;
  isActive?: boolean;
  timeZoneOffset?: string;
}

export interface INamedEntity {
  id: string;
  name: string;
  createdAt?: string;
  code?: string;
}

export interface ISelectOption {
  value: any;
  label: string;
}

export type IStepField = {
  name: string;
  required: boolean;
  computePositive?: boolean;
  fields?: IStepField[];
  requiredFunc?: (data: any) => number;
  completedFunc?: (data: any) => number;
};

export interface IStep {
  key: string;
  title: string;
  subtitle: string;
  hideControls: boolean;
  fields: IStepField[];
  order: number;
  draftFormRules?: IFormRules;
  approvalFormRules?: IFormRules;
  getDraftFormRules?: any;
  getApprovalFormRules?: any;
}

export interface IStepMap {
  [key: string]: IStep;
}

export interface IStepCompletedMap {
  [key: string]: {
    completed: number;
    required: number;
    title: string;
    order: number;
  };
}

export enum Gender {
  MALE = 0,
  FEMALE = 1,
}

export enum ToastType {
  SUCCESS = 'success',
  WARN = 'warning',
  ERROR = 'error',
  INFO = 'info',
}

export const genderMap = {
  [Gender.MALE]: 'Male',
  [Gender.FEMALE]: 'Female',
};

export const booleanOptionList = [
  { value: 1, label: 'Yes' },
  { value: 0, label: 'No' },
];

export enum TimeFilterType {
  ALL_TIMES = '',
  TODAY = 0,
  LAST_WEEK = 1,
  LAST_TWO_WEEK = 2,
  LAST_MONTH = 3,
  LAST_YEAR = 4,
}

export enum TimeFilterWorkerActivity {
  ALL_TIMES = '',
  TODAY = 0,
  LAST_WEEK = 1,
  LAST_TWO_WEEK = 2,
  LAST_MONTH = 3,
  LAST_YEAR = 4,
}

export const timeFilterWorkerActivityMap = {
  [TimeFilterWorkerActivity.ALL_TIMES]: 'All Times',
  [TimeFilterWorkerActivity.TODAY]: 'Today',
  [TimeFilterWorkerActivity.LAST_WEEK]: 'Last 7 days',
  [TimeFilterWorkerActivity.LAST_MONTH]: 'Last 30 days',
  [TimeFilterWorkerActivity.LAST_YEAR]: 'Last 365 days',
};

export enum SortingType {
  ASCENDING = 'ascending',
  DESCENDING = 'descending',
}

export const timeFilterTypeMap = {
  [TimeFilterType.ALL_TIMES]: 'All Times',
  [TimeFilterType.TODAY]: 'Today',
  [TimeFilterType.LAST_WEEK]: 'Last 7 days',
  [TimeFilterType.LAST_MONTH]: 'Last 30 days',
  [TimeFilterType.LAST_YEAR]: 'Last 365 days',
};

export const dayFilterTypeMap = {
  [TimeFilterType.LAST_WEEK]: 'Last 7 days',
  [TimeFilterType.LAST_TWO_WEEK]: 'Last 15 days',
  [TimeFilterType.LAST_MONTH]: 'Last 30 days',
};

export const timeFilterOptionList = [
  { value: TimeFilterType.ALL_TIMES, label: timeFilterTypeMap[TimeFilterType.ALL_TIMES] },
  { value: TimeFilterType.TODAY, label: timeFilterTypeMap[TimeFilterType.TODAY] },
  { value: TimeFilterType.LAST_WEEK, label: timeFilterTypeMap[TimeFilterType.LAST_WEEK] },
  { value: TimeFilterType.LAST_MONTH, label: timeFilterTypeMap[TimeFilterType.LAST_MONTH] },
  { value: TimeFilterType.LAST_YEAR, label: timeFilterTypeMap[TimeFilterType.LAST_YEAR] },
];

export const dayFilterOptionList = [
  { value: TimeFilterType.LAST_WEEK, label: dayFilterTypeMap[TimeFilterType.LAST_WEEK] },
  { value: TimeFilterType.LAST_TWO_WEEK, label: dayFilterTypeMap[TimeFilterType.LAST_TWO_WEEK] },
  { value: TimeFilterType.LAST_MONTH, label: dayFilterTypeMap[TimeFilterType.LAST_MONTH] },
];

export const workerActivityPeriodList = [
  { value: TimeFilterWorkerActivity.ALL_TIMES, label: timeFilterWorkerActivityMap[TimeFilterWorkerActivity.ALL_TIMES] },
  { value: TimeFilterWorkerActivity.TODAY, label: timeFilterWorkerActivityMap[TimeFilterWorkerActivity.TODAY] },
  { value: TimeFilterWorkerActivity.LAST_WEEK, label: timeFilterWorkerActivityMap[TimeFilterWorkerActivity.LAST_WEEK] },
  { value: TimeFilterWorkerActivity.LAST_MONTH, label: timeFilterWorkerActivityMap[TimeFilterWorkerActivity.LAST_MONTH] },
  { value: TimeFilterWorkerActivity.LAST_YEAR, label: timeFilterWorkerActivityMap[TimeFilterWorkerActivity.LAST_YEAR] },
];

export enum IJobFilter {
  TRADE = 0,
  WORK_EXPERIENCE = 1,
}

export const jobFilterMap = {
  [IJobFilter.TRADE]: 'Trade',
  [IJobFilter.WORK_EXPERIENCE]: 'Work Experience',
};

export const jobDataFilterList = [
  { value: IJobFilter.TRADE, label: jobFilterMap[IJobFilter.TRADE] },
  { value: IJobFilter.WORK_EXPERIENCE, label: jobFilterMap[IJobFilter.WORK_EXPERIENCE] },
];

export enum IDemographicFilter {
  PRIMARY_LANGUAGE = 0,
  ETHNICITY = 1,
  GENDER = 2,
}

export const demographicFilterMap = {
  [IDemographicFilter.PRIMARY_LANGUAGE]: 'Primary Language',
  [IDemographicFilter.ETHNICITY]: 'Ethnicity',
  [IDemographicFilter.GENDER]: 'Gender',
};

export const demographicFilterList = [
  { value: IDemographicFilter.ETHNICITY, label: demographicFilterMap[IDemographicFilter.ETHNICITY] },
  { value: IDemographicFilter.GENDER, label: demographicFilterMap[IDemographicFilter.GENDER] },
  { value: IDemographicFilter.PRIMARY_LANGUAGE, label: demographicFilterMap[IDemographicFilter.PRIMARY_LANGUAGE] },
];

export enum GenericType {
  NOT_DEFINED = -1,
}

export const genericType = {
  [GenericType.NOT_DEFINED]: 'Not Defined',
};

export enum IDempgrahicLanguage {
  ENGLISH = 0,
  SPANISH = 1,
  OTHER = 2,
}

export enum IDemographicEthnicity {
  ASIAN = 0,
  BLACK = 1,
  HISPANIC = 2,
  WHITE = 3,
  OTHER = 4,
}

export enum IDemographicGender {
  MALE = 0,
  FEMALE = 1,
}

export const demographicMap = {
  [IDemographicFilter.PRIMARY_LANGUAGE]: {
    ...genericType,
    [IDempgrahicLanguage.ENGLISH]: 'English',
    [IDempgrahicLanguage.SPANISH]: 'Spanish',
    [IDempgrahicLanguage.OTHER]: 'Other',
  },
  [IDemographicFilter.ETHNICITY]: {
    ...genericType,
    [IDemographicEthnicity.ASIAN]: 'Asian',
    [IDemographicEthnicity.BLACK]: 'Black',
    [IDemographicEthnicity.HISPANIC]: 'Hispanic',
    [IDemographicEthnicity.WHITE]: 'White',
    [IDemographicEthnicity.OTHER]: 'Other',
  },
  [IDemographicFilter.GENDER]: {
    ...genericType,
    [IDemographicGender.MALE]: 'Male',
    [IDemographicGender.FEMALE]: 'Female',
  },
};

export enum DateFormat {
  DATE,
  DATE_TEXT,
  DATE_MONTH,
  MONTH_YEAR,
  DATE_TIME,
  NUMERIC_DATE,
  DASHED_NUMERIC,
}

export const dateFormatMap = {
  [DateFormat.DATE]: 'MMM DD, YYYY',
  [DateFormat.DATE_TEXT]: 'MMMM DD, YYYY',
  [DateFormat.DATE_MONTH]: 'DD MMM',
  [DateFormat.MONTH_YEAR]: 'MMMM YYYY',
  [DateFormat.DATE_TIME]: 'MMM DD, YYYY hh:mm a',
  [DateFormat.NUMERIC_DATE]: 'MM/DD/YYYY',
  [DateFormat.DASHED_NUMERIC]: 'YYYY-MM-DD',
};

export enum OptionFilterType {
  NO_RELATED = 'no-related',
}

export enum ITraining {
  SAFETY_ORIENTATION = 0,
  EQUIPMENT_TRAINING = 1,
  VEHICLE_TRAINING = 2,
}

export const trainingMap = {
  ...genericType,
  [ITraining.SAFETY_ORIENTATION]: 'Safety Orientation',
  [ITraining.EQUIPMENT_TRAINING]: 'Equipment Training',
  [ITraining.VEHICLE_TRAINING]: 'Vehicle Training',
};

export const pieBlueMap = {
  0: '#00346B',
  1: '#006DF7',
  2: '#59A0FA',
  3: '#B2D3FD',
  4: '#D9E9FE',
  5: '#00346B',
};

export enum GroupValidationType {
  NOT_MANDATORY = 1,
  AT_LEAST_ONE = 2,
  ALL_MANDATORY = 3,
}

export const GroupValidationOptions = [
  { value: GroupValidationType.NOT_MANDATORY, label: 'Not mandatory' },
  { value: GroupValidationType.AT_LEAST_ONE, label: 'At least one is mandatory' },
  { value: GroupValidationType.ALL_MANDATORY, label: 'All Mandatory' },
];
