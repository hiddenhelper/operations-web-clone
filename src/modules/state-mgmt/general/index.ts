import { ReducerRegistry } from '../ReducerRegistry';
import { EpicRegistry } from '../EpicRegistry';
import { initialState } from './state';
import { actions, ActionType } from './actions';
import { reducer } from './reducer';
import { epics } from './epics';

export const generalState = { initialState, reducer, actions, ActionType, epics };

ReducerRegistry.register('general', reducer);
epics.forEach(epic => EpicRegistry.addEpic(epic));
