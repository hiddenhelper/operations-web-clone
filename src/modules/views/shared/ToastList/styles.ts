import { createStyles, makeStyles } from '@material-ui/core/styles';
import { STYLE } from '../../../../constants';
import { toREM } from '../../../../utils/generalUtils';

export const useStyles = makeStyles(theme =>
  createStyles({
    container: {
      display: 'flex',
      position: 'fixed',
      justifyContent: 'center',
      width: '100%',
      zIndex: 99999,
      top: '10px',
      flexDirection: 'column',
      alignItems: 'center',
      lineHeight: '24px',
    },
    toastItem: {
      marginTop: 15,
      padding: '0 16px',
      borderRadius: '4px',
      boxShadow: '0 0 5px 0 rgba(0,0,0,0.25)',
      color: STYLE.COLOR.WHITE,
      alignItems: 'center',
      justifyContent: 'center',
      fontSize: 15,
      fontWeight: 400,
      height: `${toREM(40)}`,
      display: 'flex',
      width: 'auto',
      '& p': {
        marginBottom: 0,
      },
    },
    success: {
      backgroundColor: STYLE.COLOR.SUCCESS,
    },
    error: {
      backgroundColor: STYLE.COLOR.ERROR,
    },
    warn: {
      backgroundColor: STYLE.COLOR.SUCCESS,
    },
    info: {
      backgroundColor: STYLE.COLOR.SUCCESS,
    },
  })
);
