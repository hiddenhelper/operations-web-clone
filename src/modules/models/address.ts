import { INamedEntity } from './general';

export enum AddressFields {
  LINE_1 = 'line1',
  LINE_2 = 'line2',
  COUNTRY = 'country',
  COUNTRY_ID = 'countryId',
  CITY = 'city',
  STATE = 'stateCode',
  STATE_NAME = 'stateName',
  ZIP_CODE = 'zipCode',
  COUNTY = 'county',
  BOROUGH = 'borough',
  LATITUDE = 'latitude',
  LONGITUDE = 'longitude',
  ATTENTION_1 = 'attention1',
  ATTENTION_2 = 'attention2',
}

export enum LocationsFields {
  NAME = 'name',
}

export enum Borough {
  MANHATTAN = 'Manhattan',
  BROOKLYN = 'Brooklyn',
  QUEENS = 'Queens',
  STATEN_ISLAND = 'Staten Island',
  THE_BRONX = 'The Bronx',
}

export interface IAddress {
  [AddressFields.LINE_1]: string;
  [AddressFields.LINE_2]: string;
  [AddressFields.CITY]: string;
  [AddressFields.STATE]: string;
  [AddressFields.ZIP_CODE]: string;
  [AddressFields.COUNTRY]?: INamedEntity;
  [AddressFields.COUNTRY_ID]?: string;
  [AddressFields.COUNTY]?: string;
  [AddressFields.ATTENTION_1]?: string;
  [AddressFields.ATTENTION_2]?: string;
  [AddressFields.STATE_NAME]?: string;
  [AddressFields.BOROUGH]?: Borough;
  [AddressFields.LATITUDE]: number;
  [AddressFields.LONGITUDE]: number;
}

export interface ILocation {
  id?: string;
  [LocationsFields.NAME]?: string;
  isUsed?: boolean;
}

export const addressFieldRules = {
  required: [
    { name: AddressFields.LINE_1, required: true },
    { name: AddressFields.LINE_2, required: false },
    { name: AddressFields.COUNTRY_ID, required: true, computePositive: true },
    { name: AddressFields.STATE, required: true },
    { name: AddressFields.ZIP_CODE, required: true },
    { name: AddressFields.CITY, required: true },
    { name: AddressFields.ATTENTION_1, required: false },
    { name: AddressFields.ATTENTION_2, required: false },
  ],
  notRequired: [
    { name: AddressFields.LINE_1, required: false },
    { name: AddressFields.LINE_2, required: false },
    { name: AddressFields.COUNTRY_ID, required: false },
    { name: AddressFields.STATE, required: false },
    { name: AddressFields.ZIP_CODE, required: false },
    { name: AddressFields.CITY, required: false },
    { name: AddressFields.ATTENTION_1, required: false },
    { name: AddressFields.ATTENTION_2, required: false },
  ],
};

export const boroughOptionList = [
  {
    value: Borough.MANHATTAN,
    label: Borough.MANHATTAN,
  },
  {
    value: Borough.BROOKLYN,
    label: Borough.BROOKLYN,
  },
  {
    value: Borough.QUEENS,
    label: Borough.QUEENS,
  },
  {
    value: Borough.STATEN_ISLAND,
    label: Borough.STATEN_ISLAND,
  },
  {
    value: Borough.THE_BRONX,
    label: Borough.THE_BRONX,
  },
];

export enum ZipCodeFormatType {
  FIVE_DIGITS = 0,
  FIVE_DIGITS_PLUS_FOUR = 1,
  CANADA_FORMAT = 2,
}

export const zipCodeFormatMap: { [key: string]: { placeholder: string; mask: any[] } } = {
  [ZipCodeFormatType.FIVE_DIGITS]: {
    placeholder: '00000',
    mask: [/\d/, /\d/, /\d/, /\d/, /\d/, /\d/],
  },
  [ZipCodeFormatType.FIVE_DIGITS_PLUS_FOUR]: {
    placeholder: '00000-0000',
    mask: [/\d/, /\d/, /\d/, /\d/, /\d/, '-', /\d/, /\d/, /\d/, /\d/],
  },
  [ZipCodeFormatType.CANADA_FORMAT]: {
    placeholder: 'ANA-NAN',
    mask: [/[a-zA-Z]/, /\d/, /[a-zA-Z]/, '-', /\d/, /[a-zA-Z]/, /\d/],
  },
};

export const getFallbackAddress = () => ({
  line1: null,
  line2: null,
  city: null,
  countryId: null,
  stateCode: null,
  zipCode: null,
  county: null,
  latitude: null,
  longitude: null,
  stateName: null,
});

export const getFallbackLocation = (): ILocation => ({
  name: null,
});
