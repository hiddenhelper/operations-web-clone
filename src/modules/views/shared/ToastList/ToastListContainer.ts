import { connect } from 'react-redux';

import { IRootState } from '../../../state-mgmt/rootState';
import ToastList from './ToastList';

export const mapStateToProps = (state: IRootState) => ({
  list: state.general.toastList,
});

export default connect(mapStateToProps, null)(ToastList);
