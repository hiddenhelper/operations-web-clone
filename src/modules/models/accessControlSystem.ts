import { ILocation } from './address';
import { IDevice, DeviceStatus } from './device';

export enum AccessControlSystemType {
  PORTAL = 0,
  TURNSTILE = 1,
  ENCLOSED_TS = 2,
  HANDHELD = 3,
}

export enum ReaderType {
  Entrance = 0,
  Exit = 1,
  BiDirectionalByAntenna = 2,
  BiDirectionalBySpec = 3,
}

export interface IReader {
  serialNumber: string;
  lastMaintenanceDate?: string;
  inServiceDate: string;
  model: string;
  notes?: string;
  hostname?: string;
  directionType?: ReaderType;
  hostAddress?: string;
  telnetConnectionPort?: string;
  sshConnectionPort?: string;
  httpConnectionPort?: string;
  tcpConnectionPort?: string;
}

export enum LifeCycleType {
  NEW = 0,
  UPGRADE = 1,
  IN_FIELD_TRANSFER = 2,
  REFURBISHED = 3,
}

export enum AccessControlSystemVersion {
  V1 = 0,
  V2 = 1,
  V3 = 2,
  V3_FACIAL = 3,
}

export interface IAccessControlSystem extends IDevice {
  type: AccessControlSystemType;
  lifeCycle: LifeCycleType;
  lastRefurbishedDate: string;
  lastMaintenanceDate: string;
  inServiceDate: string;
  warrantyExpirationDate: string;
  orderDate?: string;
  vendor?: string;
  invoice?: string;
  reader1?: IReader;
  reader2?: IReader;
  version: AccessControlSystemVersion;
  project?: {
    id: string;
    name: string;
  };
  status: DeviceStatus;
  location?: ILocation;
  hasReverseInstallation?: boolean;
  reader1SerialNumber?: string;
  reader2SerialNumber?: string;
  reader1Hostname?: string;
  reader2Hostname?: string;
}

export interface IProjectAccessControlSystem {
  id?: string;
  accessControlSystemId?: string;
  projectId?: string;
  type: AccessControlSystemType;
  serialNumber: string;
  locationId?: string;
  location?: ILocation;
  deviceName: string;
  hasReverseInstallation: boolean;
  reader1: IReader;
  reader2?: IReader;
}

export interface IProjectAccessControlSystemByLocation {
  location?: ILocation;
  accessControlSystems: IAccessControlSystem[];
}

export const accessControlSystemTypeMap = {
  [AccessControlSystemType.ENCLOSED_TS]: 'Enclosed TS',
  [AccessControlSystemType.HANDHELD]: 'Handheld',
  [AccessControlSystemType.PORTAL]: 'Open Portal',
  [AccessControlSystemType.TURNSTILE]: 'Turnstile',
};

export const typeOptionList = [
  { value: AccessControlSystemType.PORTAL, label: accessControlSystemTypeMap[AccessControlSystemType.PORTAL] },
  { value: AccessControlSystemType.TURNSTILE, label: accessControlSystemTypeMap[AccessControlSystemType.TURNSTILE] },
  { value: AccessControlSystemType.ENCLOSED_TS, label: accessControlSystemTypeMap[AccessControlSystemType.ENCLOSED_TS] },
  { value: AccessControlSystemType.HANDHELD, label: accessControlSystemTypeMap[AccessControlSystemType.HANDHELD] },
];

export const lifeCycleMap = {
  [LifeCycleType.NEW]: 'New',
  [LifeCycleType.UPGRADE]: 'Upgrade',
  [LifeCycleType.IN_FIELD_TRANSFER]: 'In Field Transfer',
  [LifeCycleType.REFURBISHED]: 'Refurbished',
};

export const lifeCycleOptionList = [
  { value: LifeCycleType.NEW, label: lifeCycleMap[LifeCycleType.NEW] },
  { value: LifeCycleType.UPGRADE, label: lifeCycleMap[LifeCycleType.UPGRADE] },
  { value: LifeCycleType.IN_FIELD_TRANSFER, label: lifeCycleMap[LifeCycleType.IN_FIELD_TRANSFER] },
  { value: LifeCycleType.REFURBISHED, label: lifeCycleMap[LifeCycleType.REFURBISHED] },
];

export const deviceVersionMap = {
  [AccessControlSystemVersion.V1]: 'V1',
  [AccessControlSystemVersion.V2]: 'V2',
  [AccessControlSystemVersion.V3]: 'V3',
  [AccessControlSystemVersion.V3_FACIAL]: 'V3Facial',
};

export const deviceVersionOptionList = [
  { value: AccessControlSystemVersion.V1, label: deviceVersionMap[AccessControlSystemVersion.V1] },
  { value: AccessControlSystemVersion.V2, label: deviceVersionMap[AccessControlSystemVersion.V2] },
  { value: AccessControlSystemVersion.V3, label: deviceVersionMap[AccessControlSystemVersion.V3] },
  { value: AccessControlSystemVersion.V3_FACIAL, label: deviceVersionMap[AccessControlSystemVersion.V3_FACIAL] },
];

export const readerTypeMap = {
  [ReaderType.Entrance]: 'Entrance',
  [ReaderType.Exit]: 'Exit',
  [ReaderType.BiDirectionalByAntenna]: 'BiDirectional By Antenna',
  [ReaderType.BiDirectionalBySpec]: 'BiDirectional By Spec',
};

export const readerTypeByTypeMap = {
  [AccessControlSystemType.PORTAL]: [
    { label: readerTypeMap[ReaderType.Entrance], value: ReaderType.Entrance },
    { label: readerTypeMap[ReaderType.Exit], value: ReaderType.Exit },
    { label: readerTypeMap[ReaderType.BiDirectionalByAntenna], value: ReaderType.BiDirectionalByAntenna },
  ],
  [AccessControlSystemType.TURNSTILE]: [{ label: readerTypeMap[ReaderType.BiDirectionalBySpec], value: ReaderType.BiDirectionalBySpec }],
  [AccessControlSystemType.ENCLOSED_TS]: [{ label: readerTypeMap[ReaderType.BiDirectionalBySpec], value: ReaderType.BiDirectionalBySpec }],
  [AccessControlSystemType.HANDHELD]: [],
};

export const getFallbackAccessControlSystem = (): IAccessControlSystem => ({
  id: null,
  type: AccessControlSystemType.PORTAL,
  serialNumber: null,
  lifeCycle: LifeCycleType.NEW,
  lastRefurbishedDate: null,
  lastMaintenanceDate: null,
  inServiceDate: null,
  warrantyExpirationDate: null,
  orderDate: null,
  vendor: null,
  invoice: null,
  project: null,
  status: null,
  notes: null,
  version: AccessControlSystemVersion.V1,
  reader1: getFallbackReader(),
  reader2: null,
  hasReverseInstallation: false,
});

export const getFallbackReader = (): IReader => ({
  serialNumber: null,
  lastMaintenanceDate: null,
  inServiceDate: null,
  model: null,
  notes: null,
});

export const getFallbackProjectAccessControlSystem = (): IProjectAccessControlSystem => ({
  type: AccessControlSystemType.PORTAL,
  accessControlSystemId: null,
  serialNumber: null,
  deviceName: null,
  hasReverseInstallation: false,
  reader1: null,
  reader2: null,
});
