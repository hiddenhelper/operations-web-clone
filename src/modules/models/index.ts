import * as GeneralModel from './general';
import * as UserModel from './user';
import * as ClientModel from './client';
import * as ProjectModel from './project';
import * as ResourceModel from './resource';
import * as WorkerModel from './worker';
import * as AccessControlSystemModel from './accessControlSystem';
import * as BadgePrintingSystemModel from './badgePrinterSystem';
import * as DeviceModel from './device';
import * as InventoryModel from './inventory';
import * as BadgeModel from './badge';
import * as CertificationModel from './certification';
import * as FileModel from './file';
import * as AddressModel from './address';
import * as ConsentFormModel from './consentForm';
import * as StatisticsModel from './statistics';
import * as TrainingModel from './training';
import * as PaymentModel from './payment';
import * as InvoiceModel from './invoice';
import * as ProcoreModel from './procore';
import * as SearchModel from './search';
import * as ProjectNewModel from './project-new';

export {
  GeneralModel,
  UserModel,
  ClientModel,
  ProjectModel,
  ResourceModel,
  AccessControlSystemModel,
  WorkerModel,
  BadgePrintingSystemModel,
  DeviceModel,
  InventoryModel,
  BadgeModel,
  CertificationModel,
  FileModel,
  AddressModel,
  ConsentFormModel,
  StatisticsModel,
  TrainingModel,
  PaymentModel,
  InvoiceModel,
  ProcoreModel,
  SearchModel,
  ProjectNewModel,
};
