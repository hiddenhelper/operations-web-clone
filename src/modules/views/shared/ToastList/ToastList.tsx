import React, { memo } from 'react';
import { ToastType } from '../../../models/general';

import { useStyles } from './styles';

export interface IToastListProps {
  list: { _id: string; message: string; type: ToastType }[];
  maxLength?: number;
}

const ToastList = ({ list, maxLength = 3 }: IToastListProps) => {
  const classes = useStyles();
  const truncatedList = [...list]
    .reverse()
    .slice(0, maxLength)
    .reverse();
  return (
    <div key="ToastList" className={classes.container}>
      {truncatedList.map(item => (
        <div key={item._id} className={`${classes.toastItem} ${classes[item.type]}`}>
          <p>{item.message}</p>
        </div>
      ))}
    </div>
  );
};

export default memo(ToastList);
