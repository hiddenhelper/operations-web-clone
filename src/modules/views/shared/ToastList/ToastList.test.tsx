import * as React from 'react';
import { render } from '@testing-library/react';

import { ToastType } from '../../../models/general';
import ToastList, { IToastListProps } from './ToastList';

describe('ToastList', () => {
  global.console.error = () => null;
  global.console.warn = () => null;
  let props: IToastListProps;
  let Component;

  beforeEach(() => {
    global.console.error = () => null;
    props = {
      list: [
        { message: 'boom', type: ToastType.ERROR, _id: '1' },
        { message: 'the blah was blah succesfully. the blah was blah succesfully', type: ToastType.SUCCESS, _id: '2' },
        { message: 'something somehting', type: ToastType.INFO, _id: '3' },
        { message: 'warn something somehting', type: ToastType.WARN, _id: '4' },
      ],
    };
    Component = render(<ToastList {...props} />);
  });

  it('should render', () => {
    expect(Component.container).toMatchSnapshot();
  });
});
