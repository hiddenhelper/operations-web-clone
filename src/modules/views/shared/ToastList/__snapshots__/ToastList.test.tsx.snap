// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToastList should render 1`] = `
<div>
  <div
    class="makeStyles-container-1"
  >
    <div
      class="makeStyles-toastItem-2 makeStyles-success-3"
    >
      <p>
        the blah was blah succesfully. the blah was blah succesfully
      </p>
    </div>
    <div
      class="makeStyles-toastItem-2 makeStyles-info-6"
    >
      <p>
        something somehting
      </p>
    </div>
    <div
      class="makeStyles-toastItem-2 undefined"
    >
      <p>
        warn something somehting
      </p>
    </div>
  </div>
</div>
`;
